### PR TITLE
fix(random): Enhance security by replacing platform-specific randomizer code with Rust's bindings

### PR DESCRIPTION
**Pull request by sergei-boiko-trustwallet**
This pull request refactors and unifies the implementation of cryptographically secure random number generation across the codebase. It removes platform-specific random number generator code (for JNI, Swift, and WASM), and replaces it with a single Rust-backed implementation that is exposed to C++ via FFI. The new approach improves code maintainability, error handling, and testability, and ensures consistent random byte generation across all platforms.

Key changes include:

**Core random number generation refactor:**

* Removed platform-specific random number generator implementations in `jni/cpp/Random.cpp`, `swift/Sources/SecRandom.m`, and `wasm/src/Random.cpp`, consolidating all random byte generation through a new Rust FFI function (`crypto_random_buffer`).
* Replaced the previous C++ `random_buffer` implementation with a wrapper that calls the Rust FFI function, and updated the API to return a status code (`RandomResult`).

**Rust FFI and memory handling:**

* Added `crypto_random_buffer` in Rust (`rust/tw_crypto/src/ffi/crypto_random.rs`), which uses `rand::thread_rng()` to fill buffers securely, and defined error codes for FFI.
* Introduced `CByteArrayMut` in Rust for safe FFI buffer handling.
* Registered the new Rust FFI module in the main FFI mod.

**Error handling and usage updates:**

* Updated all C++ code that generates random bytes (e.g., for IVs, salts, private keys) to use the new `Random::random_buffer` API and to handle errors more robustly (throwing exceptions or returning null on failure instead of aborting or terminating).
* Improved error propagation in Rust mnemonic generation to handle random generator failures explicitly.

**Testing:**

* Added comprehensive Rust FFI tests for the new `crypto_random_buffer` function, including edge cases for buffer validity and randomness.

These changes ensure a single, secure, and consistent random number generation mechanism across all supported platforms, simplify the codebase, and improve error handling for cryptographic operations.

### CODE REVIEW

File: rust/tw_memory/src/ffi/c_byte_array_mut.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    /// Returns a slice.
+    ///
+    /// # Safety
+    ///
+    /// The inner data must be valid.
+    pub unsafe fn as_mut(&mut self) -> &mut [u8] {
+        if self.data.is_null() {
+            return &mut [];
```
Comment (nikhil-gupta-tw) [inline rust/tw_memory/src/ffi/c_byte_array_mut.rs]: Seems a bit unsafe to return a reference to a local array?

File: rust/tw_memory/src/ffi/c_byte_array_mut.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// Copyright © 2017 Trust Wallet.
+
+/// A C-compatible wrapper over a mutable byte array pointer given as an FFI argument.
+#[repr(C)]
+pub struct CByteArrayMut {
```
Comment (nikhil-gupta-tw) [inline rust/tw_memory/src/ffi/c_byte_array_mut.rs]: Could we not use https://github.com/trustwallet/wallet-core/blob/master/rust/tw_memory/src/ffi/c_byte_array.rs?

If not, we should probably implement the `Drop` and free FFI APIs?

**Review by nikhil-gupta-tw — CHANGES_REQUESTED** [review]

File: rust/tw_memory/src/ffi/c_byte_array_mut.rs (hunk 1 above)
Comment (sergei-boiko-trustwallet) [reply rust/tw_memory/src/ffi/c_byte_array_mut.rs]: Right, missed that. I did the same for immutable reference, and for mutable it makes no sense

File: rust/tw_memory/src/ffi/c_byte_array_mut.rs (hunk 2 above)
Comment (sergei-boiko-trustwallet) [reply rust/tw_memory/src/ffi/c_byte_array_mut.rs]: `CByteArrayMut` doesn't own memory it was initialized with, but just a mutable reference. So there is no need to implement `Drop` for it.
Please note that `CByteArray` is designed to be used to allocate memory in Rust and "give" the ownership to C++. But `CByteArrayMut` is used to take a mutable pointer to a byte-array, and convert it to a mutable slice

Comment (nikhil-gupta-tw) [reply rust/tw_memory/src/ffi/c_byte_array_mut.rs]: Got it, thanks for the explanation. Should we name it better than? Cause currently it just feels like a mutable version of `CByteArray`?

File: rust/tw_memory/src/ffi/c_byte_array_mut.rs (hunk 1 above)
Comment (sergei-boiko-trustwallet) [reply rust/tw_memory/src/ffi/c_byte_array_mut.rs]: Done

Comment (mdmamun198710) [reply rust/tw_memory/src/ffi/c_byte_array_mut.rs]: Hello sir I need your help
